scipy/stats/tests/test_fisher_exact.py

```diff
@@ -0,0 +1,123 @@
+        assert_equal(oddsratio, 0)
+        assert_allclose(pvalue, 5/11, rtol=1e-11)
+
+    @pytest.mark.parametrize('parameters, rresult', data)
+    def test_results_from_r(self, parameters, rresult):
+        alternative = parameters.alternative.replace('.', '-')
+        result = fisher_exact(parameters.table,
+                              alternative=alternative,
+                              confidence_level=parameters.confidence_level)
+        assert_allclose(result.pvalue, rresult.pvalue, rtol=1e-11)
+        if result.conditional_odds_ratio < 400:
+            or_rtol = 5e-4
+            ci_rtol = 2e-2
+        else:
+            or_rtol = 5e-2
+            ci_rtol = 1e-1
+        assert_allclose(result.conditional_odds_ratio,
+                        rresult.conditional_odds_ratio, rtol=or_rtol)
+        assert_allclose(result.conditional_odds_ratio_ci,
+                        rresult.conditional_odds_ratio_ci, rtol=ci_rtol)
+        # Also do a self-check for the conditional odds ratio.
+        # With the computed conditional odds ratio as the noncentrality
+        # parameter of the noncentral hypergeometric distribution with
+        # parameters table.sum(), table[0].sum(), and table[:,0].sum() as
+        # total, ngood and nsample, respectively, the mean of the distribution
+        # should equal table[0, 0].
+        cor = result.conditional_odds_ratio
+        table = np.array(parameters.table)
+        total = table.sum()
+        ngood = table[0].sum()
+        nsample = table[:, 0].sum()
+        # Avoid the warning from log(cor) when cor == 0.
+        lognc = np.log(cor) if cor > 0 else -np.inf
+        nchg_mean = _nc_hypergeom_mean(lognc, total, ngood, nsample)
+        assert_allclose(nchg_mean, table[0, 0], rtol=1e-13)
+
+    @pytest.mark.slow
+    def test_large_numbers(self):
+        # Test with some large numbers. Regression test for #1401
+        pvals = [5.55976657259e-11, 2.66568479113e-11, 1.36320302448e-11]
+        for pval, num in zip(pvals, [75, 76, 77]):
+            res = fisher_exact([[17704, 496], [1065, num]])[1]
+            assert_allclose(res, pval, rtol=1e-5)
+
+        res = fisher_exact([[18000, 80000], [20000, 90000]])[1]
+        assert_allclose(res, 0.275148170406, rtol=1e-5)
+
+    def test_raises(self):
+        # test we raise an error for wrong shape of input.
+        with pytest.raises(ValueError, match='shape'):
+            fisher_exact(np.arange(6).reshape(2, 3))
+
+    @pytest.mark.parametrize('table', [
+        [[0, 0], [5, 10]],
+        [[5, 10], [0, 0]],
+        [[0, 5], [0, 10]],
+        [[5, 0], [10, 0]],
+    ])
+    def test_row_or_col_zero(self, table):
+        result = fisher_exact(table)
+        assert_equal(result.pvalue, 1.0)
+        assert_equal(result.sample_odds_ratio, np.nan)
+        assert_equal(result.conditional_odds_ratio, np.nan)
+        assert_equal(result.conditional_odds_ratio_ci, (0, np.inf))
+
+    def test_less_greater(self):
```

Comment:
Curious why not parametrize this like `test_row_or_col_zero` above?

Reply:
I forget why. I'll convert this to a parametrized test.